manim/utils/space_ops.py

```diff
@@ -299,37 +466,94 @@ def get_unit_normal(v1, v2, tol=1e-6):
 ###
 
 
-def compass_directions(n=4, start_vect=RIGHT):
+def compass_directions(n=4, start_vect=RIGHT) -> np.ndarray:
+    """Finds the cardinal directions using tau.
+
+    Parameters
+    ----------
+    n : int, optional
+        The amount to be rotated, by default 4
+    start_vect : np.ndarray, optional
+        The directon for the angle to start with, by default RIGHT
+
+    Returns
+    -------
+    np.ndarray
+        The angle which has been rotated.
+    """
     angle = TAU / n
     return np.array([rotate_vector(start_vect, k * angle) for k in range(n)])
 
 
-def complex_to_R3(complex_num):
+def complex_to_R3(complex_num: complex) -> np.ndarray:
     return np.array((complex_num.real, complex_num.imag, 0))
 
 
-def R3_to_complex(point):
+def R3_to_complex(point: Sequence[float]) -> np.ndarray:
     return complex(*point[:2])
 
 
 def complex_func_to_R3_func(complex_func):
     return lambda p: complex_to_R3(complex_func(R3_to_complex(p)))
 
 
-def center_of_mass(points):
+def center_of_mass(points: Sequence[float]) -> np.ndarray:
+    """Gets the center of mass of the points in space.
+
+    Parameters
+    ----------
+    points : Sequence[float]
+        The points to find the center of mass from.
+
+    Returns
+    -------
+    np.ndarray
+        The center of mass of the points.
+    """
     points = [np.array(point).astype("float") for point in points]
     return sum(points) / len(points)
 
 
-def midpoint(point1, point2):
-    return center_of_mass([point1, point2])
+def midpoint(
+    point1: Sequence[float], point2: Sequence[float]
+) -> Union[float, np.ndarray]:
+    """Gets the midpoint of two points.
 
+    Parameters
+    ----------
+    point1 : Sequence[float]
+        The first point.
+    point2 : Sequence[float]
+        The second point.
 
-def line_intersection(line1, line2):
+    Returns
+    -------
+    Union[float, np.ndarray]
+        The midpoint of the points
     """
-    Returns intersection point of two lines,
-    each defined with a pair of vectors determining
-    the end points
+    return center_of_mass([point1, point2])
+
+
+def line_intersection(line1: Sequence[float], line2: Sequence[float]) -> np.ndarray:
```

Comment:
Shouldn't the lines be a list of np.ndarray?

Reply:
Can't it be both?